api_app/connectors_manager/connectors/misp.py

```diff
@@ -1,10 +1,94 @@
 # This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
 # See the file 'LICENSE' for copying permission.
 
+from api_app import helpers
+import pymisp
 
-from ..classes import Connector
+from api_app.connectors_manager.classes import Connector
+
+
+INTELOWL_MISP_TYPE_MAP = {
+    "ip": "ip-src",
+    "domain": "domain",
+    "url": "url",
+    # "hash" (checked from helpers.get_hash_type)
+    "general": "text",  # misc field, so keeping text
+    "file": "filename",
+}
 
 
 class MISP(Connector):
+    def set_params(self, params):
+        self.ssl_check = params.get("ssl_check", True)
+        self.debug = params.get("debug", False)
+        self.__url_name = self._secrets["url_key_name"]
+        self.__api_key = self._secrets["api_key_name"]
+
+    @property
+    def _event_obj(self):
+        obj = pymisp.MISPEvent()
+        obj.info = f"Intelowl Analysis: {self._job.observable_name}"
+        obj.distribution = 0  # your_organisation_only
+        obj.threat_level_id = 4  # undefined
+        obj.analysis = 2  # completed
+        obj.add_tag("intelowl")
+
+        # Add tags from Job
+        for tag in self._job.tags.all():
+            obj.add_tag(f"{tag.label}:{tag.color}")
+
+        return obj
+
+    @property
+    def _base_attr_obj(self):
+        if self._job.is_sample:
+            type = "file"
+            value = f"{self._job.file_name} ({self._job.file_mimetype})"
+        else:
+            type = self._job.observable_classification
+            value = self._job.observable_name
+            if type == "hash":
+                matched_type = helpers.get_hash_type()
+                # override with matched type
+                type = matched_type if matched_type is not None else "text"
+
+        obj = pymisp.MISPAttribute()
+        obj.type = INTELOWL_MISP_TYPE_MAP[type]
+        obj.value = value
+        obj.add_tag(f"intelowl-{type}")
+
+        return obj
+
+    @property
+    def _link_attr_obj(self):
+        """
+        Returns attribute linking analysis on IntelOwl instance
+        """
+        obj = pymisp.MISPAttribute()
+        obj.type = "link"
+        # TODO: get instance url
+        obj.value = f"http://localhost/pages/scan/result/{self.job_id}"
```

Comment:
TODO

Reply:
I can't do much about this, does it make sense to ask for it in the config?

```json
...
config: {
    "intelowl_instance_url": "http://myintelowl.com"
}


```

Reply:
Considering that this is more a "project" config rathen than a "connector" config, could it make sense to add that value as a variable in `settings.py` and extract the configuration from the `env_file_app` if set? 
Then we could set a default like "localhost" if it is not set

Reply:
Yep, that's better, should have thought of that.